src/extension.ts

```diff
@@ -120,11 +120,26 @@ export function activate(context: ExtensionContext) {
   context.subscriptions.push(commands.registerTextEditorCommand('magit.stage-file', Command.primeFileCommand(stageFile)));
   context.subscriptions.push(commands.registerTextEditorCommand('magit.unstage-file', Command.primeFileCommand(unstageFile)));
 
-  context.subscriptions.push(commands.registerTextEditorCommand('magit.dispatch', Command.primeRepo(async (repository: MagitRepository) => {
-    const uri = DispatchView.encodeLocation(repository);
-    views.set(uri.toString(), new DispatchView(uri));
-    return workspace.openTextDocument(uri).then(doc => window.showTextDocument(doc, { viewColumn: MagitUtils.showDocumentColumn(), preview: false }));
-  }, false)));
+  context.subscriptions.push(commands.registerCommand('magit.dispatch', async () => {
+    const editor = window.activeTextEditor;
+    const repository = await MagitUtils.getCurrentMagitRepo(editor?.document.uri);
+
+    if (repository) {
+      // Checks for existing Magit dispatch view
+      for (const [uri, view] of views ?? []) {
+        if (view instanceof DispatchView) {
+          if (editor?.document.uri.path === DispatchView.UriPath) {
+            return;
+          }
+          return workspace.openTextDocument(view.uri).then(doc => window.showTextDocument(doc, { viewColumn: MagitUtils.showDocumentColumn(), preview: false }));
+        }
+      }
+
```

Comment:
```suggestion
```
I'd like to propose removing this bit. Doesn't feel like its pullings its weight considering how simple the `DispatchView` view is and there being no update logic for it. (needsUpdate = false)

Reply:
Yeap, I removed and force pushed.